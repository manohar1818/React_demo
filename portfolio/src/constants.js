export const HOME="Home";
export const ABOUT="About";
export const CONTACT="Contact";
export const GALLERY="Gallery";
export const NAME="Welcome All , I'm Manohar Reddy";
export const NAME_LABEL="Name";
export const EMAIL_LABEL="Email";
export const MESSAGE_LABEL="Message";
export const GALLERY_ADMIN="GalleryAdmin";
export const URL_LABEL="Url";
export const ABOUT_ME=" I am Manohar Reddy from Mahatma Gandhi Institute of Technology. I usually love learning new Technologies . My Hobies playing Badminton ";
export const TECHNICAL_SKILLS="Technical Skills";
export const JAVA="Java";
export const DBMS="DBMS";
export const SPRING="Spring";
export const SPRING_BOOT="Spring boot";
export const REST="REST";
export const NETWORKING="Networking";
export const LINUX="Linux";
export const HTML_CSS="HTML,CSS";
export const PROJECTS="Projects";
export const GATE_PASS_MANAGEMENT_SYSTEM="Gate Pass Management System";
export const MAKING_DATA_BASED_DESCIONS_USING_GOOGLE_ANALYTICS="Making Data Based Decisions Using Google Analytics";
export const ONLINE_DOCTOR_APPOINTMENT="Online Doctor Appointment";
export const SPRING_BOOT_REST_API_MOVIES="Spring boot REST API which gives information about movies and associated actors";
export const COURSES_PROJECT="Spring boot application which gives information about courses and related topics";
export const DETECTION_PREDICTION_CLOUDS="Detection and Prediction of Clouds";
export const REACT="React";
export const CONNECT="connect with me";
export const SUBMIT="submit";
export const ADD="Add";
export const ADD_IMAGE="Add Image";
export const LOGOUT="Logout";
export const DESIGNATION="Full Stack Developer Intern";
export const SAVE="save";
export const LOGIN="login";
export const ERROR_PAGE="Error Page";
export const LOGIN_FIRST="You have to login first";
export const BACK_TO_LOGIN_PAGE="Back to login page";
export const LOGIN_PAGE="LoginPage";
export const INVALID_NAME="invalid name";
export const INVALID_EMAIL="invalid email";
export const INVALID_MESSAGE="invalid message";
export const INVALID_URL="invalid url";
export const LOADING="Loding";
