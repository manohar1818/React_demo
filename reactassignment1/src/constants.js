export const HOME="Home";
export const ABOUT="About";
export const CONTACT="Contact";
export const GALLERY="Gallery";
export const NAME="Welcome All, I'm Manohar Reddy";
export const NAME_LABEL="Name";
export const EMAIL_LABEL="Email";
export const MESSAGE_LABEL="Message";
export const GALLERY_ADMIN="GalleryAdmin";
export const URL_LABEL="Url";
export const ABOUT_ME=" I am Manohar Reddy from Mahatma Gandhi Institute of Technology.I'm passionate about coding and learning new technologies.In free time I like to chill at home watching movies and colouring .I love to be surrounded by positive and motivating people. ";
export const TECHNICAL_SKILLS="Technical Skills";
export const PERSONAL="Personal details";
export const MYNAME = 'Name : Gangidi Manohar Reddy';
export const CONTACTNO = "Contact-no : 7995520569";
export const EMAIL = "Email : gmr1818@gmail.com";
export const DOB = "DOB : 18 th June 1998";
export const LANGUAGES = "Languages : English , Hindi , Telugu";
export const HOBBIES ="Hobbies : Real Estate , Badminton";
export const JAVA="Java";
export const DBMS="DBMS";
export const SPRING="Python";
export const SPRING_BOOT="Spring boot";
export const REST="Machine Learning";
export const NETWORKING="Networking";
export const LINUX="Linux";
export const HTML_CSS="HTML,CSS";
export const PROJECTS="Projects";
export const a="Crop yield usig ML";
export const b="College Management System";
export const c="Sports App";
export const d="Smart Parking System";
export const REACT="React";
export const CONNECT="connect with me";
export const SUBMIT="submit";
export const ADD="Add";
export const ADD_IMAGE="Add Image";
export const LOGOUT="Logout";
export const DESIGNATION="Intern at Zemoso Technologies";
export const SAVE="save";
export const LOGIN="login";
export const ERROR_PAGE="Error Page";
export const LOGIN_FIRST="You have to login first";
export const BACK_TO_LOGIN_PAGE="Back to login page";
export const LOGIN_PAGE="LoginPage";
export const INVALID_NAME="invalid name";
export const INVALID_EMAIL="invalid email";
export const INVALID_MESSAGE="invalid message";
export const INVALID_URL="invalid url";
export const LOADING="Loding";
