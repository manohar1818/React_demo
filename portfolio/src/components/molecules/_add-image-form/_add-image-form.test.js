import React from 'react';
import { create } from "react-test-renderer";
import { render ,fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import AddImageForm from './_add-image-form';
import * as Constants from "../../../constants";
//const name="image1";
const nameError="";
const urlError="";
const error=false;

describe("AddImageForm component",() =>{
    test("Matches the snapshot",() => {
        const  addImageForm= create(<AddImageForm urlError={urlError} nameError={nameError} error={error} />);
        expect(addImageForm.toJSON()).toMatchSnapshot();

    })
    test("Matches the snapshot when error is present",() => {
        const  addImageForm= create(<AddImageForm urlError="invalid url" nameError="invalid name" error="True" />);
        expect(addImageForm.toJSON()).toMatchSnapshot();

    })
    test("AddImageForm element is rendered",() => {
        const { getByTestId, getByText } = render(<AddImageForm urlError={urlError} nameError={nameError} error={error} />);
        expect(getByTestId("addImageForm")).toBeInTheDocument();
        
    });
    
  
   

});