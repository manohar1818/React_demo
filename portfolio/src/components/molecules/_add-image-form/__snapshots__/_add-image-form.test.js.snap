// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddImageForm component Matches the snapshot 1`] = `
<div
  data-testid="addImageForm"
>
  <h3
    className="MuiTypography-root makeStyles-heading-3 MuiTypography-h3"
  >
    Add Image
  </h3>
  <form
    className="makeStyles-form-1"
  >
    <div
      className="makeStyles-input-2"
    >
      <div
        data-testid="input"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5"
        >
          Name
        </h5>
        <div
          className="makeStyles-styleInputFields-5"
          data-testid="inputField"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline"
            data-testid="inputAtom"
            onClick={[Function]}
            required={true}
            style={
              Object {
                "color": "#000",
              }
            }
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              data-testid="name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
          </div>
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-4 MuiTypography-body1"
      />
    </div>
    <div
      className="makeStyles-input-2"
    >
      <div
        data-testid="input"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5"
        >
          Url
        </h5>
        <div
          className="makeStyles-styleInputFields-5"
          data-testid="inputField"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline"
            data-testid="inputAtom"
            onClick={[Function]}
            required={true}
            style={
              Object {
                "color": "#000",
              }
            }
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              data-testid="url"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
          </div>
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-4 MuiTypography-body1"
      />
    </div>
    <div
      className="makeStyles-button-6"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            Add
          </span>
        </span>
      </button>
    </div>
  </form>
</div>
`;

exports[`AddImageForm component Matches the snapshot when error is present 1`] = `
<div
  data-testid="addImageForm"
>
  <h3
    className="MuiTypography-root makeStyles-heading-3 MuiTypography-h3"
  >
    Add Image
  </h3>
  <form
    className="makeStyles-form-1"
  >
    <div
      className="makeStyles-input-2"
    >
      <div
        data-testid="input"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5"
        >
          Name
        </h5>
        <div
          className="makeStyles-styleInputFields-5"
          data-testid="inputField"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline"
            data-testid="inputAtom"
            onClick={[Function]}
            required={true}
            style={
              Object {
                "color": "#000",
              }
            }
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              data-testid="name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
          </div>
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-4 MuiTypography-body1"
      >
        invalid name
      </p>
    </div>
    <div
      className="makeStyles-input-2"
    >
      <div
        data-testid="input"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5"
        >
          Url
        </h5>
        <div
          className="makeStyles-styleInputFields-5"
          data-testid="inputField"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline"
            data-testid="inputAtom"
            onClick={[Function]}
            required={true}
            style={
              Object {
                "color": "#000",
              }
            }
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              data-testid="url"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
          </div>
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-4 MuiTypography-body1"
      >
        invalid url
      </p>
    </div>
    <div
      className="makeStyles-button-6"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
        data-testid="button"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            Add
          </span>
        </span>
      </button>
    </div>
  </form>
</div>
`;
